File: Algoritmo_2/Pi.java
package Algoritmo_2;

import java.util.Scanner;

public class Pi {

    static double potencia(int base, int expoente) {
        return Math.pow(base, expoente);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int s;
        double somador = 0;
        double auxiliar = 0;
        double num = 1;
        int expoente = 3;

        System.out.print("Digite um número inteiro: ");
        s = scanner.nextInt();

        double numerador = 1;
        double denominador = 1;

        for (int i = 1; i <= s; i++) {
            auxiliar = (numerador / potencia((int)denominador, expoente)) * num;

            num *= -1;
            denominador += 2;
            somador += auxiliar;
        }

        System.out.printf("%.5f\n", somador);
    }
}

File: Algoritmo_2/pi.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

double potencia(int base, int expoente) {
    return pow(base, expoente);
}

int main() {

    int s; 
    double somador = 0;
    double auxiliar = 0;
    double num = 1;
    int expoente = 3;

    scanf("%d",&s);

    double numerador = 1; // + 1
    double denominador = 1; // x * 2 + 1 = y

    for (int i = 1; i <= s; i++) {
        auxiliar = (numerador / potencia(denominador, expoente)) * num;

        num = num * (-1);
        denominador = denominador + 2;
        somador = somador + auxiliar;
    }   

    printf("%.5f\n", somador);

    return 0;
}